### DISCUSSION TITLE
Include timestamp in directory server logs

### DISCUSSION THREAD

**Discussion by strk** [jamulussoftware/jamulus#3001] [Ideas]
It would be useful to have a timestamp in these messages:
> Requested to register entry for ...
> Requested to unregister entry for ...

It's generally weird to see timestamp in some messages but not in others.
This is similar to #277 but for server DIRECTORY logs

**Comment by pljones** [comment]
If you use systemd, then journalctl will list the log lines with timestamps.
```
$ sudo journalctl -u Jamulus@Directory3 --since 00:00 | tail -20
...
Jan 27 18:21:17 fs-peter Jamulus-Directory3[1074]: Requested to register entry for xx.xx.xx.xx:22124 (xx.xx.xx.xx:22124): tkap-jam-net (external)
Jan 27 18:22:32 fs-peter Jamulus-Directory3[1074]: Requested to register entry for xx.xx.xx.xx:22124 (xx.xx.xx.xx:22124): Bohemia Jazz Club (external)
Jan 27 18:22:51 fs-peter Jamulus-Directory3[1074]: Requested to register entry for xx.xx.xx.xx:20100 (xx.xx.xx.xx:20100): 🎶 Jazz Open Jam (external)
```

If you're on Windows, you may be able to convince Qt logging to route to the Windows system log.  But I've not investigated that.

**Reply by strk** [reply]
I'm not using systemd, just starting the directory manually when I see a need for it

**Reply by strk** [reply]
what I'm saying is it's still inconsistent to have timestamp on some logs and not on others, or I guess you get double-timestamp for the client-to-server connection logs, with systemd, right ?

**Reply by pljones** [reply]
It's generally down to the log capture system to timestamp lines on receipt.

You might find answers here: https://unix.stackexchange.com/questions/26728/prepending-a-timestamp-to-each-line-of-output-from-a-command

**Reply by strk** [reply]
Yes I know how to do that, so yes, can accept to do it, but logs from #277 will have a double timestamp, basically

**Reply by pljones** [reply]
Why?  The file output wouldn't go to the screen.  You're trying to capture screen output that Jamulus doesn't write to a file, as far as I've understood your original post.

**Reply by strk** [reply]
Here's what I mean:

> Feb 02 11:30:03 my.server.dns jamulus[32295]: 2023-02-02 11:30:03, client.ip.address.redacted, connected (1)

One timestamp is added by syslog (Feb 02 11:30:30) while the other timestamp is written by Jamulus itself (2023-02-02 11:30:03), since it was added to respond to #277 -- it's inconsistent to have the Jamulus-added timestamp, as there's no Jamulus-added timestamp on the "Requested to register entry for.." logs instead.

**Reply by pljones** [reply]
Why are you using syslog to log the content of the Jamulus server log file?  Just use the option to write it a file?  (If you want it logged to syslog as well `tail -Ff jamulus.log | sed -e 's/^[^,]*, /'` should do.)  (The `-Ff` means you can delete the file from time to time to save disk space, if that's a problem.)

**Reply by strk** [reply]
"why are you using syslog" ? Because you suggested that !
Here: https://github.com/orgs/jamulussoftware/discussions/3001#discussioncomment-4799691
And here: https://github.com/orgs/jamulussoftware/discussions/3001#discussioncomment-4832713

I filed a ticket for the inconsistency: https://github.com/jamulussoftware/jamulus/issues/3004

**Reply by pljones** [reply]
For the directory, which goes only from Jamulus to sysout/syserr, you need to add additional system logging.

For the server, where it has its own log file, you don't need to add additional system logging.

I'm asking why you want the server logging to syslog.

**Comment by strk** [comment]
In any case this thread can be closed as resolved, if that makes sense (I've no idea how "github discussions" work)

**Reply by ann0see** [reply]
As it's not in a q&a category, we don't mark any answer as final answer -> no closing needed.
